modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java

```diff
@@ -580,15 +657,40 @@ private NavigationMenuItem _toNavigationMenuItem(
 							"getSiteSitePage", contextUriInfo,
 							arguments.toArray(new Object[0]));
 					});
-				setUseCustomName(
+
+				setContentURL(
 					() -> {
-						if (layout == null) {
+						if ((layout != null) ||
+							StringUtil.equals(type, "url") ||
+							type.contains("asset") ||
+							StringUtil.equals(type, "navigationMenu")) {
+
 							return null;
 						}
 
-						return Boolean.valueOf(
-							unicodeProperties.getProperty(
-								"useCustomName", "false"));
+						if (Objects.equals(type, FileEntry.class.getName())) {
+							type = DLFileEntry.class.getName();
+						}
+
+						List<Object> arguments = new ArrayList<>();
+
+						Class<?> clazz = _getClassNameContent(
+							type,
+							GetterUtil.getLong(
+								unicodeProperties.getProperty("classPK")));
+
+						String methodName = _getMethodName(
+							type,
+							GetterUtil.getLong(
+								unicodeProperties.getProperty("classPK")));
+
+						Long contentId = _getItemId(unicodeProperties);
+
+						arguments.add(contentId);
+
+						return JaxRsLinkUtil.getJaxRsLink(
+							"headless-delivery", clazz, methodName,
```

Comment:
Here is where we could infer the base path instead of put `headless-delivery` so we could generate the link for other navigation menu items that don't belong to `headless-delivery`.